framework/sandstone_opts.cpp

```diff
@@ -0,0 +1,845 @@
+                break;
+
+            case max_test_count_option:
+                app->max_test_count = ParseIntArgument<>{"--max-test-count"}();
+                break;
+
+            case max_test_loop_count_option:
+                app->max_test_loop_count = ParseIntArgument<>{"--max-test-loop-count"}();
+                if (app->max_test_loop_count == 0)
+                        app->max_test_loop_count = std::numeric_limits<int>::max();
+                break;
+
+                /* deprecated options */
+            case longer_runtime_option:
+            case max_concurrent_threads_option:
+            case mem_sample_time_option:
+            case mem_samples_per_log_option:
+            case no_mem_sampling_option:
+            case no_triage_option:
+            case schedule_by_option:
+            case shortened_runtime_option:
+            case triage_option:
+            case weighted_testrun_option:
+                warn_deprecated_opt(long_options[coptind].name);
+                break;
+
+            case 0:
+                /* long option setting a value */
+                continue;
+            default:
+                suggest_help(argv);
+                return EX_USAGE;
+            }
+        }
+    } else {
+        // Default options for the simplified OpenDCDiag cmdline
+        static struct option restricted_long_options[] = {
+            { "help", no_argument, nullptr, 'h' },
+            { "query", no_argument, nullptr, 'q' },
+            { "service", no_argument, nullptr, 's' },
+            { "version", no_argument, nullptr, version_option },
+            { nullptr, 0, nullptr, 0 }
+        };
+
+        while ((opt = simple_getopt(argc, argv, restricted_long_options)) != -1) {
+            switch (opt) {
+            case 'q':
+                // ### FIXME
+                fprintf(stderr, "%s: --query not implemented yet\n", argv[0]);
+                abort();
+            case 's':
+                // keep in sync above
+                app->endtime = MonotonicTimePoint::max();
+                app->service_background_scan = true;
+                break;
+            case version_option:
+                opts.action = Action::version;
+                return EXIT_SUCCESS;
+            case 'h':
+                usage(argv);
+                opts.action = Action::exit;
+                return opt == 'h' ? EXIT_SUCCESS : EX_USAGE;
+            default:
+                suggest_help(argv);
+                opts.action = Action::exit;
+                return EX_USAGE;
+            }
+        }
+
+        if (SandstoneConfig::NoLogging) {
+            app->shmem->output_format = SandstoneApplication::OutputFormat::no_output;
+        } else  {
+            app->shmem->verbosity = 1;
+        }
+
+        app->delay_between_tests = 50ms;
+        app->thermal_throttle_temp = INT_MIN;
+        opts.fatal_errors = true;
+        opts.builtin_test_list_name = "auto";
```

Comment:
Question: should we move the test file processing here, from sandstone.cpp?

That would remove the need to have `builtin_test_list_name`, `enable_tests` and such in `opts`.

Reply:
this is more of a processing of the parsed opts, so I don't think this is a good idea. We could isolate it into a separate function though. `main` would then become even more concise. But this is not a scope of this PR.